src/Optimization/hiopAlgFilterIPM.cpp

```diff
@@ -1055,7 +1055,7 @@ hiopSolveStatus hiopAlgFilterIPMQuasiNewton::run()
     /************************************************
      * update mu and other parameters
      ************************************************/
-    while(_err_log<=kappa_eps * _mu) {
+    if(_err_log<=kappa_eps * _mu) {
       //update mu and tau (fraction-to-boundary)
       bret = updateLogBarrierParameters(*it_curr, _mu, _tau, _mu, _tau);
       if(!bret) break; //no update is necessary
```

Comment:
why do we change this `while` loop to `if`? I think `while` is designed to reduce `mu` faster.
In addition, if we change to `if`,  the `break` command on line 1061 may break the outer loop and give us unpredictable result.

Reply:
under `elastic_mode` using `while` tighten up bounds too quickly and enters FR quite often. Also, aggressive reduction puts a lot of stress on the linear solver for some reason. Not sure using `while` for the non`elastic_mode` outweigh the, but I can leave the `while` on for nonelastic mode.

Will remove fix the `break`.